tree/tree/test/BulkApiMultiple.cxx

```diff
@@ -25,27 +29,261 @@ class BulkApiMultipleTest : public ::testing::Test {
 
       // Otherwise, we keep with the current ROOT defaults.
       auto tree = new TTree("T", "A ROOT tree of floats.");
-      float f = 2;
-      double g = 3;
+
+      float f = fFloatInitial;
+      double g = fDoubleInitial;
+      bool b = true;
+      short s  = 4;
+      int i = 5;
+      Long64_t ll = (1<<20) + 1;
+      char c = 7;
+
       TBranch *branch2 = tree->Branch("myFloat", &f, 320000, 1);
       TBranch *branch3 = tree->Branch("myDouble", &g, 320000, 1);
+      TBranch *branch4 = tree->Branch("myBool", &b, 320000, 1);
+      TBranch *branch5 = tree->Branch("myShort", &s, 320000, 1);
+      TBranch *branch6 = tree->Branch("myInt", &i, 320000, 1);
+      TBranch *branch7 = tree->Branch("myLongLong", &ll, 320000, 1);
+      TBranch *branch8 = tree->Branch("myChar", &c, 320000, 1);
+
       branch2->SetAutoDelete(kFALSE);
       branch3->SetAutoDelete(kFALSE);
+      branch4->SetAutoDelete(kFALSE);
+      branch5->SetAutoDelete(kFALSE);
+      branch6->SetAutoDelete(kFALSE);
+      branch7->SetAutoDelete(kFALSE);
+      branch8->SetAutoDelete(kFALSE);
+
       for (Long64_t ev = 0; ev < fEventCount; ev++) {
          tree->Fill();
+         if (ev && (ev % gRollOver) == 0) {
+            f = 1.0;
+            g = 2.0;
+         }
          f ++;
          g ++;
+         b = !b;
+         s ++;
+         i ++;
+         ll ++;
+         c ++;
       }
       hfile = tree->GetCurrentFile();
       hfile->Write();
-      tree->Print();
+      //tree->Print();
       printf("Successful write of all events.\n");
       hfile->Close();
 
       delete hfile;
    }
 };
 
+
+template <typename T>
+bool rollover(Long64_t /* ev */, T & /* value */)
+{
+   return false;
+}
+
+bool rollover(Long64_t ev, float &value)
+{
+   if (ev && (ev % gRollOver) == 0) {
+      value = BulkApiMultipleTest::fFloatInitial;
+      return true;
+   }
+   return false;
+}
+
+bool rollover(Long64_t ev, double &value)
+{
+   if (ev && (ev % gRollOver) == 0) {
+      value = BulkApiMultipleTest::fDoubleInitial;
+      return true;
+   }
+   return false;
+}
+
+template <typename T>
+void increment(Long64_t ev, T &value)
+{
+   if (!rollover(ev, value))
+      ++value;
+}
+
+void increment(Long64_t /* ev */, bool &value)
+{
+   value = !value;
+}
+
+template <typename T>
+bool compare(T value, T expected)
+{
+   return (ULong64_t)value == (ULong64_t)expected;
```

Comment:
That can still fail the comparison `(ULong64_t)14.999998 == (ULong64_t)15.000001`. As you use the same increments for filling and testing (consecutive `++`) it might be okay...

Reply:
Yes, the same increment strategy is assumed/required.